crates/bevy_pbr/src/light.rs

```diff
@@ -368,44 +368,30 @@ pub struct Clusters {
 }
 
 impl Clusters {
-    fn new(tile_size: UVec2, screen_size: UVec2, z_slices: u32, near: f32, far: f32) -> Self {
+    fn new(screen_size: UVec2, dimensions: UVec3, near: f32, far: f32) -> Self {
+        debug_assert!(screen_size.x > 0 && screen_size.y > 0);
+        debug_assert!(dimensions.x > 0 && dimensions.y > 0 && dimensions.z > 0);
         let mut clusters = Self {
-            tile_size,
+            tile_size: UVec2::ONE,
             axis_slices: Default::default(),
             near,
             far,
             aabbs: Default::default(),
             lights: Default::default(),
         };
-        clusters.update(tile_size, screen_size, z_slices);
+        clusters.update(screen_size, dimensions);
         clusters
     }
 
-    fn from_screen_size_and_dimensions(
-        screen_size: UVec2,
-        dimensions: UVec3,
-        near: f32,
-        far: f32,
-    ) -> Self {
-        debug_assert!(screen_size.x > 0 && screen_size.y > 0);
-        debug_assert!(dimensions.x > 0 && dimensions.y > 0 && dimensions.z > 0);
-        Clusters::new(
-            (screen_size.as_vec2() / dimensions.xy().as_vec2())
-                .ceil()
-                .as_uvec2(),
-            screen_size,
-            dimensions.z,
-            near,
-            far,
-        )
-    }
-
-    fn update(&mut self, tile_size: UVec2, screen_size: UVec2, z_slices: u32) {
+    fn update(&mut self, screen_size: UVec2, cluster_dimensions: UVec3) {
```

Comment:
i think the per-frame updates of ``clusters.near`` and ``clusters.far`` have been dropped; they need to be updated as they can change and they're used in render/light.rs to calculate matching cluster_factors there.

Reply:
Ooops yeah thats very important. Nice save!